bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v1/form/HeaderFooterTextImpl.java

```diff
@@ -0,0 +1,106 @@
+ ~ you may not use this file except in compliance with the License.
+ ~ You may obtain a copy of the License at
+ ~
+ ~     http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~ Unless required by applicable law or agreed to in writing, software
+ ~ distributed under the License is distributed on an "AS IS" BASIS,
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+package com.adobe.cq.forms.core.components.internal.models.v1.form;
+
+import javax.annotation.PostConstruct;
+import javax.inject.Inject;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.models.annotations.Default;
+import org.apache.sling.models.annotations.Exporter;
+import org.apache.sling.models.annotations.Model;
+import org.apache.sling.models.annotations.injectorspecific.InjectionStrategy;
+import org.apache.sling.models.annotations.injectorspecific.ValueMapValue;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import com.adobe.aemds.guide.utils.GuideUtils;
+import com.adobe.cq.export.json.ComponentExporter;
+import com.adobe.cq.export.json.ExporterConstants;
+import com.adobe.cq.forms.core.components.util.AbstractComponentImpl;
+import com.adobe.cq.forms.core.components.util.ComponentUtils;
+import com.adobe.cq.wcm.core.components.models.Text;
+import com.adobe.cq.wcm.core.components.models.datalayer.ComponentData;
+import com.adobe.cq.wcm.core.components.models.datalayer.builder.DataLayerBuilder;
+import com.day.cq.wcm.api.WCMMode;
+
+@Model(
+    adaptables = { SlingHttpServletRequest.class, Resource.class },
+    adapters = { Text.class, ComponentExporter.class },
+    resourceType = { HeaderFooterTextImpl.RESOURCE_TYPE_V1, HeaderFooterTextImpl.RESOURCE_TYPE_V2 })
+@Exporter(name = ExporterConstants.SLING_MODEL_EXPORTER_NAME, extensions = ExporterConstants.SLING_MODEL_EXTENSION)
+
+public class HeaderFooterTextImpl extends AbstractComponentImpl implements Text {
+
+    protected static final String RESOURCE_TYPE_V1 = "core/wcm/components/text/v1/text";
+    protected static final String RESOURCE_TYPE_V2 = "core/wcm/components/text/v2/text";
+
+    @ValueMapValue(injectionStrategy = InjectionStrategy.OPTIONAL)
+    @Nullable
+    private String text;
+
+    @ValueMapValue(injectionStrategy = InjectionStrategy.OPTIONAL)
+    @Default(booleanValues = false)
+    private boolean textIsRich;
+
+    @Inject
+    private Resource resource;
+
+    @PostConstruct
+    private void initModel() {
+        if (request != null && i18n == null) {
+            i18n = GuideUtils.getI18n(request, resource);
+        }
+    }
+
+    @Override
+    public String getText() {
```

Comment:
Ideally only this function should be overridden and implemented, all the other functions should be extended. Can you check and fic this ?

Reply:
@rismehta Sure, I have extended those functions present in AbstractComponentImpl. But for the other function they have to defined here, as the translate function and i18n definition are present in AbstractComponentImpl, the impl is extedning it. So from wcm we are just using there interface and all the function definitions need to copied.